src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -22,42 +22,27 @@ internal abstract class AbstractConvertConcatenationToInterpolatedStringRefactor
     {
         public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
-            // Currently only supported if there is no selection.  We could consider relaxing
-            // this if the selection is of a string concatenation expression.
-            if (context.Span.Length > 0)
-            {
-                return;
-            }
-
             var cancellationToken = context.CancellationToken;
```

Comment:
overall i like this change.  however, my preference is:

if we allow non-empty spans, we require that the span start/end can't be in the *middle* of a token.  don't want someone making a random textual selection, but then getting these sorts of features.  it feels pretty confusing as to what the selection/refactoring then mean IMO.

Reply:
@CyrusNajmabadi I understand your preference, however, I chose to do it this way because the other 'Convert to interpolated string' refactoring works the same way. If they behave differently (as they were doing before this change), that's also confusing.

I.e. when you have: 
`string.Format("Hello {0} there {1}", 1, 2);` 
you can also have a selection, either in full or in part, and the refactoring will be offered.

I'd be happy to change it back to not allowing selections if you prefer it that way though. Please let me know.

Reply:
My personal preference is that we should continually tring to make things better, even if that leads to periods of inconsistency.  :)   I personally believe it would be better in *both* cases to not support this  functionality if there is a selection that splits individual tokens.  However, that doesn't mean i don't want to fix one case just because another case exists.  Always be improving, and we'll continually reach better and better places.  If we dont' improve things because of other issues elsewhere, we commonly reach local minimums :-/

Reply:
@CyrusNajmabadi You don't need to convince me, I completely agree! Just wanted to make sure you knew the difference in the similar refactoring :). I can't make the changes today anymore, I hope I can get around to it tomorrow.

Reply:
I disallowed selections again now.